module/zfs/zfs_vnops.c

```diff
@@ -2607,6 +2607,12 @@ zfs_getattr_fast(struct inode *ip, struct kstat *sp)
 
 	mutex_enter(&zp->z_lock);
 
+	bzero(sp, sizeof (*sp));
+#if defined(HAVE_STAT_RESULT_MASK)
+	/* Must be set before the call to generic_fillattr() */
+	stat->result_mask = STATX_BASIC_STATS;
+#endif /* defined(HAVE_STAT_RESULT_MASK) */
+
```

Comment:
I don't think we need this chunk, this is already done in vfs layer.
http://elixir.free-electrons.com/source/fs/stat.c?v=4.11-rc2#L67